File: parent-337/pinyougou-web-manager/src/main/webapp/js/service/brandservice.js
// 自定义服务  可以引入其他服务 比如$http
app.service('brandService', function ($http) {
    // 查询所有品牌
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    };

    // 分页查询品牌
    this.fingPage = function (pageNum, pageSize) {
        return $http.get('../brand/findPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize);
    };

    // 根据id查询品牌
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id)
    };

    // 新建品牌的保存品牌操作
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity)
    };

    // 修改品牌的保存品牌操作
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity)
    };

    // 删除品牌
    this.del = function (ids) {
        return $http.get('../brand/del.do?ids=' + ids)
    };

    // 条件查询
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../brand/search.do?pageNum=' + pageNum + '&pageSize=' + pageSize, searchEntity)
    };

    // 查询id和name,存入一个map集合 并将多个这样的集合存入list集合中
    this.selectOptionList = function () {
        return $http.get('../brand/selectOptionList.do');
    }
});

File: parent-337/pinyougou-interface/src/main/java/cn/ithcast/core/service/BrandService.java
package cn.ithcast.core.service;

import cn.itcast.core.pojo.good.Brand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    /**
     * 查询所有品牌,并显示在浏览器
     *
     * @return
     */
    List<Brand> findAll();

    /**
     * 分页展示
     *
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findPage(Integer pageNum, Integer pageSize);

    /**
     * 新增品牌
     *
     * @param brand
     */
    void add(Brand brand);

    /**
     * 修改品牌之数据回显
     *
     * @param id
     * @return
     */
    Brand findOne(Long id);

    /**
     * 修改品牌之修改品牌
     *
     * @param brand
     */
    void update(Brand brand);


    /**
     * 删除品牌
     *
     * @param ids
     */
    void del(Long[] ids);

    /**
     * 条件查询
     */
    PageResult search(Integer pageNum, Integer pageSize, Brand brand);

    /**
     * 查询id和name,存入一个map集合 并将多个这样的集合存入list集合中
     */
    List<Map> selectOptionList();
}
